lib/cfn-model/util/wildcard_patterns.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Create array of wildcard patterns for a given input string
+
+def wildcard_patterns(input, pattern_types: %w[front back both])
+  input_string = input.to_s
+  results = [input_string]
+  pattern_types.each do |pattern_type|
+    case pattern_type
+    when 'front'
+      results += wildcard_front(input_string)
+    when 'back'
+      results += wildcard_back(input_string)
+    when 'both'
+      results += wildcard_front_back(input_string)
+    else
+      raise "no pattern of type: #{pattern_type}. Use one or more of: front, back, both"
+    end
+  end
+  results + ['*']
+end
+
+private
+
+def wildcard_back(input_string, results = [], prepend: '')
+  return results if input_string.empty?
+
+  results << "#{prepend}#{input_string}*"
+  wildcard_back(input_string.chop, results, prepend: prepend)
+end
+
+def wildcard_front(input_string, results = [])
```

Comment:
can we remove this code from cfn_nag after we push this?

Reply:
Yep!  After `wildcard_patterns.rb` and associated tests are pushed with cfn-model, I can file a PR to modify the few places in cfn_nag to require it from cfn-model.